optimum/exporters/tasks.py

```diff
@@ -659,6 +661,37 @@ def determine_framework(model: str, framework: str = None) -> str:
 
         return framework
 
+    @staticmethod
+    def infer_task_from_model(model_name_or_path):
+        tasks_to_automodels = {}
+        class_name_prefix = ""
+        if is_torch_available():
+            tasks_to_automodels = TasksManager._TASKS_TO_AUTOMODELS
+        else:
+            tasks_to_automodels = TasksManager._TASKS_TO_TF_AUTOMODELS
+            class_name_prefix = "TF"
+
+        inferred_task_name = None
+        is_local = os.path.isdir(model_name_or_path)
+
+        if is_local:
+            # TODO: implement this.
+            raise NotImplementedError("Cannot infer the task from a local directory yet.")
+        else:
+            model_info = huggingface_hub.model_info(model_name_or_path)
+            transformers_info = model_info.transformersInfo
+            if transformers_info is None or transformers_info.get("auto_model") is None:
+                raise RuntimeError(f"Could not infer the task from the model repo {model_name_or_path}")
+            auto_model_class_name = f"{class_name_prefix}{transformers_info['auto_model']}"
```

Comment:
i think `auto_model` can start with `TF` already in some cases so you just only prepend it if it's not already there